jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantWidget/ExtendedVariantWidget.js

```diff
@@ -31,71 +33,323 @@ export default jbrowse => {
+                                </div>
+                                <div className={classes.fieldValue}>
+                                    {feat["INFO"][displays[display].properties[property]]}
+                                </div>
+                            </div>
+                        )
+                    }else if(feat["INFO"][displays[display].properties[property]].length > 1){
+                        var children = []
+                        for(var val in feat["INFO"][displays[display].properties[property]]){
+                            children.push(
+                                <div className={classes.fieldSubValue}>
+                                    {feat["INFO"][displays[display].properties[property]][val]}
+                                </div>
+                            )
+                        }
+                        tempProp.push(
+                            <div className={classes.field}>
+                                <div className={classes.fieldName}>
+                                    {displays[display].properties[property]}
+                                </div>
+                                {children}
+                            </div>
+                        )
+                    }
                 }
             }
-            propertyJSX.push(tempProp)
+            if(tempProp.length != 0){
+                propertyJSX.push(tempProp)
+            }
         }
-
         var displayJSX = []
-        for(var i = 0; i < displays.length; i++){
+        for(var i = 0; i < propertyJSX.length; i++){
             displayJSX.push(
                 <BaseCard title={displays[i].name}>
-                   <div style={{ padding: '7px', width: '100%', maxHeight: 600, overflow: 'auto' }}>
-                        <Table className={displays[i].name}>
-                            <TableHead>
-                                {propertyJSX[i]}
-                            </TableHead>
-                        </Table>
-                   </div>
+                    {propertyJSX[i]}
                 </BaseCard>
             )
         }
         return displayJSX
     }
 
+    function makeChart(samples, ref, alt, classes){
+        const [state, setState] = useState(null)
+        useEffect(() => {
+            setState(
+                <BaseCard title="Genotypes">
+                    <div>
+                    Loading genotypes...
+                    </div>
+                </BaseCard>
+            )
+            var alleleCounts = {}
+            var alleleTotal = 0
+            alleleCounts[ref] = 0
+            for(var i in alt){
+                alleleCounts[alt[i]] = 0
+            }
+            var gtCounts = {}
+            var gtTotal = 0
+            for(var sample in samples){
+                var gt = samples[sample]["GT"]
+                for(var entry in gt){
+                    // CASES
+                    // ./. -- no call
+                    if(gt[entry] == "./."){
```

Comment:
this is a question: in JS === seems preferred to ==. in reality they are functionally equal here. what's best practice in react?

Reply:
Default to `===`, usually. Either works in this case due to the string comparison, but the behavior can be dangerous in other cases.

```
const number = 1234 
const stringNumber = '1234'  

console.log(number == stringNumber) //true
console.log(number === stringNumber)  //false
```

Reply:
Will keep this in mind for future development. Warrants a change in this PR? @bbimber

Reply:
Is there any reason this is more complicated than a find/replace? In src\client\JBrowse\Browser, I see a bunch of 'var' in ExtendedVariantWidget.js, and then only about 2-3 others. A switch should be super easy, unless there are cases where we need 'var' over const. I dont know typescript well enough to answer scoping, but maybe @hextraza could comment.

Reply:
wrt var/let/const, there could be logic that relies on `var`'s scoping behavior, but even if that's the case it probably shouldn't. `var` is essentially global scope, so there could be some unforeseen consequences of changing everything to `let` if we were relying on `var`'s global behavior, but it's probably pretty easy to run the module/tests and check if everything still works